pandas/core/groupby/generic.py

```diff
@@ -889,9 +889,23 @@ def aggregate(self, func=None, *args, **kwargs):
                     result = self._aggregate_multiple_funcs(
                         [func], _level=_level, _axis=self.axis
                     )
-                except AssertionError:
-                    raise
-                except Exception:
+                except ValueError as err:
+                    if "no results" not in str(err):
+                        # raised directly by _aggregate_multiple_funcs
+                        raise
+                    result = self._aggregate_frame(func)
+                except NotImplementedError as err:
+                    if "axis other than 0 is not supported" in str(err):
+                        # raised directly by _aggregate_multiple_funcs
+                        pass
+                    elif "decimal does not support skipna=True" in str(err):
+                        # FIXME: kludge for DecimalArray tests
```

Comment:
What is the end goal of this change? 
I don't think we should have special cases in the code for our tests?

Reply:
The upcoming fix here is to have DecimalArray._reduce implement support `skipna=True`.

A lot of spaghetti code here is driven by the numeric_only behavior that suppresses exceptions.